operator/src/main/java/oracle/kubernetes/operator/helpers/RollingHelper.java

```diff
@@ -208,31 +208,18 @@ public NextAction apply(Packet packet) {
 
       LOGGER.info(MessageKeys.ROLLING_SERVERS, dom.getDomainUid(), servers, readyServers);
 
-      int countToRestartNow = Math.max(1, countReady - dom.getMinAvailable(clusterName));
+      int countToRestartNow = countReady - dom.getMinAvailable(clusterName);
       Collection<StepAndPacket> restarts = new ArrayList<>();
       for (int i = 0; i < countToRestartNow; i++) {
-        restarts.add(new StepAndPacket(new RestartOneClusteredServerStep(servers, null), packet));
+        StepAndPacket serverToRestart = servers.poll();
+        if (serverToRestart != null) {
+          restarts.add(serverToRestart);
+        }
       }
-      return doForkJoin(getNext(), packet, restarts);
-    }
-  }
-
-  private static class RestartOneClusteredServerStep extends Step {
-    private final Queue<StepAndPacket> servers;
-
-    public RestartOneClusteredServerStep(Queue<StepAndPacket> servers, Step next) {
-      super(next);
-      this.servers = servers;
-    }
 
-    @Override
-    public NextAction apply(Packet packet) {
-      StepAndPacket serverToRestart = servers.poll();
-      if (serverToRestart != null) {
-        Collection<StepAndPacket> col = Collections.singleton(serverToRestart);
-        return doForkJoin(this, packet, col);
-      }
-      return doNext(packet);
+      return restarts.isEmpty()
```

Comment:
Please avoid magic numbers in the code - constants are clearer. In addition, nested ternaries can be hard to read. This would be easier to follow as a set of if's, choosing the simplest case first, such as:

```
    if (!restarts.isEmpty()) {
        doForkJoin(this, packet, restarts);
    } else if (!servers.isEmpty()) {
        doDelay(this, packet, DELAY_IN_SECONDS, TimeUnit.SECONDS);
    } else {
        doNext(packet);
    }
```

Reply:
Updated with your suggestion to make it easier to read.